packages/presentation/src/useStatus.ts

```diff
@@ -1,35 +1,59 @@
 import type {Status, StatusEvent} from '@sanity/comlink'
 import {useCallback, useMemo, useState} from 'react'
 
+/**
+ * A hook that manages and returns the connection status of multiple channels
+ *
+ * @returns {[string, (event: StatusEvent) => void]} - An array containing the
+ * current status and a function to update the status based on incoming events
+ *
+ * The status can be one of the following:
+ * - 'connected': If any channel is connected
+ * - 'connecting': If the first connection is being established
+ * - 'reconnecting': If a reconnection is in progress
+ * - 'idle': If no connections have been made yet
+ *
+ * The function to update the status takes a `StatusEvent` object which includes
+ * the channel and the status
+ */
 export function useStatus(): [string, (event: StatusEvent) => void] {
+  // State to keep track of the status of each channel
   const [statusMap, setStatusMap] = useState(
     new Map<string, {status: Status; hasConnected: boolean}>(),
   )
 
+  // Memoized computation of the overall status based on the status of individual channels
   const status = useMemo(() => {
     const values = Array.from(statusMap.values())
+    // If any channel is connected, return the `connected` status
```

Comment:
This was moved up so that the status is always "connected" if at least one channel is currently connected. This is to prevent the Preview component from re-rendering each time a popup is created.